lib/api/workloadendpoint.go

```diff
@@ -15,42 +15,107 @@
 package api
 
 import (
-	. "github.com/tigera/libcalico-go/lib/api/unversioned"
-	. "github.com/tigera/libcalico-go/lib/net"
+	"github.com/tigera/libcalico-go/lib/api/unversioned"
+	"github.com/tigera/libcalico-go/lib/net"
 )
 
+type WorkloadEndpoint struct {
+	unversioned.TypeMetadata
+	Metadata WorkloadEndpointMetadata `json:"metadata,omitempty"`
+	Spec     WorkloadEndpointSpec     `json:"spec,omitempty"`
+}
+
+// WorkloadEndpointMetadata contains the Metadata for a WorkloadEndpoint resource.
 type WorkloadEndpointMetadata struct {
-	ObjectMetadata
-	Name           string            `json:"name,omitempty" validate:"omitempty,name"`
-	WorkloadID     string            `json:"workloadID,omitempty" valid:"omitempty,name"`
-	OrchestratorID string            `json:"orchestratorID,omitempty" valid:"omitempty,name"`
-	Hostname       string            `json:"hostname,omitempty" valid:"omitempty,name"`
-	Labels         map[string]string `json:"labels,omitempty" validate:"omitempty,labels"`
+	unversioned.ObjectMetadata
+
+	// The name of the endpoint.
+	Name string `json:"name,omitempty" validate:"omitempty,name"`
+
+	// The name of the workload.
+	WorkloadID string `json:"workloadID,omitempty" valid:"omitempty,name"`
+
+	// The name of the orchestrator.
+	OrchestratorID string `json:"orchestratorID,omitempty" valid:"omitempty,name"`
+
+	// The name of the node.
+	Hostname string `json:"hostname,omitempty" valid:"omitempty,name"`
+
+	// The labels applied to the workload endpoint.  It is expected that many endpoints share
+	// the same labels. For example, they could be used to label all “production” workloads
+	// with “deployment=prod” so that security policy can be applied to production workloads.
+	Labels map[string]string `json:"labels,omitempty" validate:"omitempty,labels"`
 }
 
+// WorkloadEndpointMetadata contains the specification for a WorkloadEndpoint resource.
 type WorkloadEndpointSpec struct {
-	IPNetworks    []IPNet  `json:"ipNetworks,omitempty" validate:"omitempty"`
-	Profiles      []string `json:"profiles,omitempty" validate:"omitempty,dive,name"`
-	InterfaceName string   `json:"interfaceName,omitempty" validate:"omitempty,interface"`
-	MAC           MAC      `json:"mac,omitempty" validate:"omitempty"`
+	// IPNetworks is a list of subnets allocated to this endpoint. IP packets will only be
```

Comment:
We currently set this to be "active" in the client, and don't expose this on the public facing API.